db/column_family.h

```diff
@@ -419,6 +419,9 @@ class ColumnFamilyData {
 
   // if the database was opened with 2pc enabled
   bool allow_2pc_;
+
+  // Memtable id to track flush.
+  uint64_t last_memtable_id_;
```

Comment:
It's better to make it atomic and do fetch_add(). ConstructNewMemtable() is not supposed to call inside DB mutex, so if it's not too expensive, I would make it thread-safe.

Or I prefer having the ID set in SetMemtable() instead. That is inside the DB mutex and is easier to reason about the number is not decreasing.

Reply:
I briefly checked and thought ConstructNewMemtable will be inside DB mutex, but if not then definitely I will move it to SetMemtable().